AirCasting/SearchAndFollow/SearchLocation/SearchViewModel.swift

```diff
@@ -2,13 +2,32 @@
 //
 
 import Foundation
+import CoreLocation
+import Resolver
 
 class SearchViewModel: ObservableObject {
+    @Injected private var networkChecker: NetworkChecker
     
     @Published var isLocationPopupPresented = false
-    @Published var location = ""
+    @Published var addressName = ""
+    @Published var addresslocation = CLLocationCoordinate2D(latitude: 20.0, longitude: 20.0)
+    @Published var alert: AlertInfo?
     
     func textFieldTapped() { isLocationPopupPresented.toggle() }
     
-    func updateLocation(using newLocation: String) { location = newLocation}
+    func updateLocationName(using newLocationName: String) {
+        addressName = newLocationName
+    }
+    
+    func updateLocationAddress(using newLocationAddress: CLLocationCoordinate2D) {
+        addresslocation = newLocationAddress
+    }
+    
+    func onAppearAction(onEnd: @escaping () -> Void) {
```

Comment:
I have no clue what this method is doing 🤷‍♂️ please try to structure it a bit better